src/components/board/helpers/move-task.helper.ts

```diff
@@ -0,0 +1,70 @@
+import { DraggableLocation } from 'react-beautiful-dnd';
+import update from 'immutability-helper';
+import { FullColumnDto, UpdateTaskDto } from '~/common/types/types';
+import { TaskUpdatePayload } from '~/store/task/common';
+import { orderTasks } from './order-tasks.helper';
+
+export const moveTask = (
+  columns: Array<FullColumnDto>,
+  boardId: string,
+  source: DraggableLocation,
+  target: DraggableLocation,
+): [Array<FullColumnDto>, TaskUpdatePayload | null] => {
+  const sourceId = source.droppableId;
+  const targetId = target.droppableId;
+  const sourceIndex = columns.findIndex((column) => column.id === sourceId);
+  const targetIndex = columns.findIndex((column) => column.id === targetId);
+  const sourceColumn = columns[sourceIndex];
+  const sourceTask = sourceColumn.tasks[source.index];
+  const targetTask = sourceColumn.tasks[target.index];
+
+  if (!targetTask || !sourceTask) {
+    return [columns, null];
+  }
+
+  let result: Array<FullColumnDto>;
+
+  if (sourceIndex !== targetIndex) {
+    result = update(columns, {
+      [sourceIndex]: {
+        tasks: { $splice: [[source.index, 1]] },
+      },
+      [targetIndex]: {
+        tasks: { $splice: [[target.index, 0, sourceTask]] },
+      },
+    });
+  } else {
+    result = update(columns, {
+      [sourceIndex]: {
+        tasks: {
+          $splice: [
+            [source.index, 1],
+            [target.index, 0, sourceTask],
+          ],
+        },
+      },
+    });
+  }
+
+  const indexes = [sourceIndex, targetIndex];
+
+  result = indexes.reduce((acc, index) => orderTasks(acc, index), result);
```

Comment:
If we assign to result this in all cases, why do we need to assign something to it above? It looks like it does't make sense and we can just use const